src/protocol/templates/pingmessage.h.in

```diff
@@ -126,6 +126,6 @@ public:
         msgData[7] = 0;
     }
 
-    void set_id(uint16_t id) { memcpy((msgData + 4 + 0), &id, 2); }
+    void set_id(const uint16_t id) const { memcpy((msgData + 4 + 0), &id, 2); }
```

Comment:
why declare the parameter const if we are changing the value?

Reply:
why declare the function as const if we are modifying the object?

Reply:
This method does not change msgData pointer.

Reply:
Check pointers and references: https://en.wikipedia.org/wiki/Const_(computer_programming)

Reply:
I expected to at least find some warnings like this, but I don't see it in the job logs. What do you think?
```
warning: passing argument 1 of ‘memcpy’ discards ‘const’ qualifier from pointer target type [enabled by default]
```

Reply:
This is a normal thing in cost functions. You can also do msgData[1] = 2, this is totally ok.

Reply:
I understand how c and the compiler are happy with it now, thank you for the clarification. 

Still, I think it is not **semantically correct** to label the function const. Someone reads the header and the const qualifier tells them that no members change, however we are changing a member in this set_* function (via pointer operations). Maybe my idea is not in agreement with common practices, if so let me know.